pkg/networkservice/common/mechanisms/kernel/utils.go

```diff
@@ -20,21 +22,27 @@ import (
 	"fmt"
 	"net/url"
 
-	"github.com/networkservicemesh/api/pkg/api/networkservice"
 	kernelmech "github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/nanoid"
+)
+
+const (
+	ifPrefix = "nsm"
 )
 
 var netNSURL = (&url.URL{Scheme: "file", Path: "/proc/thread-self/ns/net"}).String()
 
-// getNameFromConnection - returns a name computed from networkservice.Connection 'conn'
-func getNameFromConnection(conn *networkservice.Connection) string {
-	ns := conn.GetNetworkService()
-	nsMaxLength := kernelmech.LinuxIfMaxLength - 5
-	if len(ns) > nsMaxLength {
-		ns = ns[:nsMaxLength]
+// generateInterfaceName - returns a random interface name with "nsm" prefix
+func generateInterfaceName() (string, error) {
```

Comment:
Let's add a comment about the calculated probability of a name collision. It might be useful for folks who are interested in the guarantees and safety of NSM.